src/azure-cli-core/azure/cli/core/auth/identity.py

```diff
@@ -307,7 +307,14 @@ def __init__(self, secret_store):
         self._entries = []
 
     def load_entry(self, sp_id, tenant):
+        from azure.cli.core._profile import env_var_auth_configured, load_env_var_credential
         self._load_persistence()
+
+        # If no login data, look for service principal credential in environment variables
+        if not self._entries and env_var_auth_configured():
+            logger.warning("Using service principal credential configured in environment variables.")
```

Comment:
Unconditionally showing warnings can breaks pipelines which enables `failOnStderr` (https://github.com/Azure/azure-cli/issues/18372).

Reply:
Sorry, I'm not sure what you mean. How should I make this warning shown "conditionally"?

Reply:
> Sorry, I'm not sure what you mean. How should I make this warning shown "conditionally"?

Well. This warning shouldn't be printed at all, as it doesn't really qualify as a warning.

Reply:
Understood. It's been changed to a `logger.debug` call.